[18.0][MIG] account_invoice_production_lot: Migration to 18.0

mb-ife opened the pull request.

eLBati committed:
[ADD] account_invoice_production_lot
copied from https://code.launchpad.net/~agilebg/account-invoicing/adding_account_invoice_production_lot_7

Commit:
Launchpad automatic translations update.

hurrinico committed:
[FIX] account_invoice_production_lot: Changed Dependences + Fixed Tests + Remove name

archetipo committed:
[MIG] account_invoice_production_lot: Migration to 8.0

* create decorator
* porting model to API 8.0
* tests
* report inovoice, adding SN on invoice report
* html formatting in formatted_note
* handled all type of invoicing policy
* description in README
* QWeb report

pedrobaeza committed:
[IMP] account_invoice_production_lot: Improve README

OCA Transbot updated translations from Transifex

cubells committed:
[MIG] account_invoice_production_lot: Migration to 10.0

pedrobaeza committed:
[FIX] account_invoice_production_lot: Don't duplicate S/N

On using the module, it results that an invoice with 2 lines, each
of them having a serial number, prints on the report both numbers
on each line. This technique has been the proper for avoiding the
problem.

OCA Transbot updated translations from Transifex

eLBati committed:
IMP account_invoice_production_lot showing quantities by lots, when tracking by lot

tafaRU committed:
[IMP] account_invoice_production_lot: unit test + improve README

* Add unit tests to cover the following use case scenario.

  Steps to reproduce:

  * create a new product 'Product Test': stockable, Invoicing Policy as Delivered quantities, and tracking by lots
  * update its quantity to 3 with new lot 'Lot1'
  * update its quantity to 3 with new lot 'Lot2'
  * sale 4 of 'Product Test', confirm the SO
  * on the delivery, split the lots and set To Do quantities as 3 for Lot1 and leave 0 for Lot2
  * validate the delivery and create a backorder

  Current behavior:

  the field 'Production Lots' of the invoice generated is filled with both lots and, consequently,
  the invoice report will also contain info related to the lot not delivered yet.

  Expected behavior:

  the invoice report only contains info related to the lot involved in the delivery, which is Lot1.

* Improve README contents and regenerate it by using oca-gen-addon-readme

pedrobaeza committed:
[MIG] account_invoice_production_lot: Migration to 11.0

* Standard procedure
* Adapt code to new stock data model
* Adapt tests
* Fix README layout
* Remove coding lines in .py

AdriaGForgeFlow committed:
[12.0][MIG] account_invoice_production_lot module

sergio-teruel committed:
[MIG] account_invoice_production_lot: Migration to v12

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-account_invoice_production_lot/

[UPD] Update account_invoice_production_lot.pot

joao-p-marques committed:
[IMP] account_invoice_production_lot: black, isort, prettier

joao-p-marques committed:
[MIG] account_invoice_production_lot: Migration to 13.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoice-reporting-13.0/account-invoice-reporting-13.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-13-0/account-invoice-reporting-13-0-account_invoice_production_lot/

[UPD] Update account_invoice_production_lot.pot

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 87.5% (7 of 8 strings)

Translation: account-invoice-reporting-13.0/account-invoice-reporting-13.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-13-0/account-invoice-reporting-13-0-account_invoice_production_lot/es/

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: account-invoice-reporting-13.0/account-invoice-reporting-13.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-13-0/account-invoice-reporting-13-0-account_invoice_production_lot/es/

fredzamoabg committed:
[MIG] account_invoice_production_lot: Migration to 14.0

[UPD] Update account_invoice_production_lot.pot

[UPD] README.rst

pedrobaeza committed:
[FIX] account_invoice_production_lot: Don't populate empyt S/N

If a product doesn't track lots, there are stock.move.line, but without
lot, and the report crashed before this commit due to getting a False as
lot name.

TT34868

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (8 of 8 strings)

Translation: account-invoice-reporting-14.0/account-invoice-reporting-14.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-14-0/account-invoice-reporting-14-0-account_invoice_production_lot/de/

victoralmau committed:
[MIG] account_invoice_production_lot: Migration to 15.0

TT37267

Commit:
[UPD] Update account_invoice_production_lot.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoice-reporting-15.0/account-invoice-reporting-15.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-15-0/account-invoice-reporting-15-0-account_invoice_production_lot/

Commit:
Translated using Weblate (Catalan)

Currently translated at 100.0% (7 of 7 strings)

Translation: account-invoice-reporting-15.0/account-invoice-reporting-15.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-15-0/account-invoice-reporting-15-0-account_invoice_production_lot/ca/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: account-invoice-reporting-15.0/account-invoice-reporting-15.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-15-0/account-invoice-reporting-15-0-account_invoice_production_lot/es/

aktiv-heli-kantawala committed:
[MIG] account_invoice_production_lot: Migration to 16.0

Commit:
[UPD] Update account_invoice_production_lot.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoice-reporting-16.0/account-invoice-reporting-16.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-16-0/account-invoice-reporting-16-0-account_invoice_production_lot/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 83.3% (5 of 6 strings)

Translation: account-invoice-reporting-16.0/account-invoice-reporting-16.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-16-0/account-invoice-reporting-16-0-account_invoice_production_lot/hr/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoice-reporting-16.0/account-invoice-reporting-16.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-16-0/account-invoice-reporting-16-0-account_invoice_production_lot/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoice-reporting-16.0/account-invoice-reporting-16.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-16-0/account-invoice-reporting-16-0-account_invoice_production_lot/it/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoice-reporting-16.0/account-invoice-reporting-16.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-16-0/account-invoice-reporting-16-0-account_invoice_production_lot/pt_BR/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoice-reporting-16.0/account-invoice-reporting-16.0-account_invoice_production_lot
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-16-0/account-invoice-reporting-16-0-account_invoice_production_lot/sv/

ng-ife commented on account_invoice_production_lot/i18n/pt_BR.po:
Why is this changed?

ng-ife commented on account_invoice_production_lot/i18n/sv.po:
this seems to be a downgrade

victoralmau committed:
[FIX] account_invoice_production_lot: Fallback to load CoA

Since odoo/odoo@d0342c8, the default existing company is not getting a
CoA automatically, provoking than the current tests fail with the error:

odoo.exceptions.UserError: No journal could be found in company My Company (San Francisco) for any of those types: sale

Thus, we put tests post-install for being sure localization modules are
installed, the same as AccountTestInvoicingCommon does, but we don't
inherit from it, as it creates an overhead creating 2 new companies and
loading their CoA and some more stuff, while we don't need all of that.

Besides, if you don't have `l10n_generic_coa` installed, you can't use
another CoA (like `l10n_es`) easily, so we put little code to select the
first available CoA.

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] account_invoice_production_lot: pre-commit auto fixes

Commit:
[MIG] account_invoice_production_lot: Migration to 18.0
